arch/x86/core/fatal.c

```diff
@@ -24,6 +24,37 @@
 
 __weak void _debug_fatal_hook(const NANO_ESF *esf) { ARG_UNUSED(esf); }
 
+#if defined(CONFIG_EXCEPTION_STACK_TRACE)
+struct stack_frame {
+	struct stack_frame *next;
+	u32_t ret_addr;
+	u32_t args;
+};
+
+static void unwind_stack(u32_t base_ptr)
+{
+	struct stack_frame *frame = (struct stack_frame *)base_ptr;
```

Comment:
It's undefined behavior to create a pointer that might be misaligned.  Might be better to make the `next` member in `struct stack_frame` a u32_t as well and only create a pointer to that if the value is aligned.